Compilation Error on Windows with VC++

```
Collatable.cc
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xlocale(337) : warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
C:\couchbase-lite-java\libraries\couchbase-lite-java-forestdb\vendor\cbforest\CBForest\Collatable.cc(77) : error C2039: 'max' : is not a member of 'std'
C:\couchbase-lite-java\libraries\couchbase-lite-java-forestdb\vendor\cbforest\CBForest\Collatable.cc(77) : error C3861: 'max': identifier not found
C:\couchbase-lite-java\libraries\couchbase-lite-java-forestdb\vendor\cbforest\CBForest\Collatable.cc(141) : warning C4068: unknown pragma
C:\couchbase-lite-java\libraries\couchbase-lite-java-forestdb\vendor\cbforest\CBForest\Collatable.cc(375) : warning C4068: unknown pragma

slice.cc
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xlocale(337) : warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
C:\couchbase-lite-java\libraries\couchbase-lite-java-forestdb\vendor\cbforest\CBForest\slice.cc(25) : error C2059: syntax error : ':'
C:\couchbase-lite-java\libraries\couchbase-lite-java-forestdb\vendor\cbforest\CBForest\slice.cc(27) : error C2059: syntax error : ':'
```

--- a/CBForest/Collatable.cc
+++ b/CBForest/Collatable.cc
@@ -19,6 +19,7 @@
 #include "Error.hh"
 #include <sstream>
 #include <iomanip> // std::setprecision
+#include <algorithm> // std::max for MSVC
 
 namespace forestdb {
 

--- a/CBForest/slice.cc
+++ b/CBForest/slice.cc
@@ -21,10 +21,14 @@ namespace forestdb {
 
     int slice::compare(slice b) const {
         // Optimized for speed
-        if (this->size < b.size)
-            return memcmp(this->buf, b.buf, this->size) ?: -1;
-        else if (this->size > b.size)
-            return memcmp(this->buf, b.buf, b.size) ?: 1;
+        if (this->size < b.size) {
+            int ret = memcmp(this->buf, b.buf, this->size);
+            return ret ? ret : -1;
+        }
+        else if (this->size > b.size) {
+            int ret = memcmp(this->buf, b.buf, b.size);
+            return ret ? ret : 1;
+        }
         else
             return memcmp(this->buf, b.buf, this->size);
     }

